src/guides/v2.3/extension-dev-guide/routing.md

```diff
@@ -350,6 +350,75 @@ As a result, by accessing the `http://site.com/learning` route, the `http://site
 
 ![Routing Result]({{ site.baseurl }}/common/images/routing-result.png)
 
+## Declaring the new custom no route processor
+
+If the standard 404 page should be extended or the customer should be directed to another page, for example, the search page where the query path will be query content, the Magento allows adding a custom no route processor.
+In order to add a new no route processor, add the argument to the ``NoRouteHandlerList``  in the ``di.xml`` file:
```

Comment:
Let's add a hint what `di.xml` should be used in this case. There's a global one, so we need to clarify to avoid confusion.

```suggestion
In order to add a new no route processor, add the argument to the ``NoRouteHandlerList``  in the ``di.xml`` file of your module:
```

Thank you.